### DISCUSSION TITLE
locking operation을 모두 unlock 앞에 두면 왜 serializability를 보장할 수 있는 것일까요?

### DISCUSSION THREAD

**Discussion by Hanjaemo** [Hanjaemo/DB-Study#19] [Q&A]
2PL protocol은 locking operation들을 모두 unlock 전에 수행하도록 구현하는 방식입니다. 이는 단 한번이라도 unlock되는 순간 더 이상 lock을 획득할 수 없기 때문에 serializability를 보장한다고 하는데요. locking operation들이 모두 unlock 앞에 있다는 것과 serializable의 상관 관계는 무엇일까요?

**Comment by gomudayya** [comment]
이게 참 저도 알듯 말듯 이해가 완벽하게 안되네요

그냥 제가 정리한것은 locking operation이 모두 unlock 앞에 있으면 

"데드락은 발생할지언정 이상현상은 발생하지 않는다." 정도로만 넘어갔습니다..

예시를 보면 그냥 그렇게되네.. 하고 이해는 되는데 일반화해서 정리를 못하겠네요.
